guava/src/com/google/common/collect/RegularImmutableSortedSet.java

```diff
@@ -162,9 +162,7 @@ public boolean equals(@Nullable Object object) {
     if (SortedIterables.hasSameComparator(comparator, that)) {
       Iterator<?> otherIterator = that.iterator();
       try {
-        Iterator<E> iterator = iterator();
-        while (iterator.hasNext()) {
-          Object element = iterator.next();
+        for (E element : this) {
```

Comment:
Haha, fair enough!

Seems IntelliJ isn't _quite_ as infallible as I thought. 😋 And I also agree to some extent that consistent, parallel iterator calls are better than replacing one of them with a for loop, so I'm happy with this.